examples/rust/Sum_over_array.rs

```diff
@@ -1,13 +1,3 @@
-pub fn sum_array_loop(input: &[i32]) -> i32 {
```

Comment:
I liked that you could see how both functions diffed in this one example. Any reason to now show it anymore? (Not too familiar with rust though, so forgive me for my ignorance)

Reply:
Yeah idk, it's pretty unidiomatic to use indexing in Rust. The linters lint against it as well. So I'd want to avoid people looking at this and think that it's idiomatic and generates similarly good ASM anyway (which it probably does in this case, but not in general).

Reply:
I have no strong opinions either way so I'll merge like this and if needed. it could be added back at a later date if someone misses it. Thank you so much for taking the time to make this changes :)